test/mri/ruby/test_module.rb

```diff
@@ -588,7 +588,7 @@ def test_s_constants
     c2 = Module.constants
     assert_equal([:WALTER], c2 - c1)
 
-    assert_equal([], Module.constants(true))
+    assert_equal([:JRUBY_PRIVATE], Module.constants(true))
```

Comment:
this is actually something that shows the implementation is leaking into all modules.
understand it might be demotivating but this raises a :red_circle: for the proposed changes ...

Reply:
Since the tests aren't passing yet, this obviously isn't ready for review though.

Reply:
Hey Andy, what I meant is what was mentioned in comments already.
The changes should NOT affect plain-old Ruby modules in any way ...
Meaning all existing Ruby tests/specs should pass as before (wout changes).

Will hold off any reviews, was just a hint to not chase a dead end potentially.